lib/client.js

```diff
@@ -379,6 +377,8 @@ Client.prototype.query = function (config, values, callback) {
         rejectOut = reject
       })
       query.callback = (err, res) => err ? rejectOut(err) : resolveOut(res)
+    } else {
+      result = query
```

Comment:
This shouldn’t be added.

Reply:
Well, it was removed intentionally and doesn’t work with promises. I’m not sure what the right fix is, though… requiring the `Query` object to be passed explicitly like in the tests? Or a separate method?

Reply:
My idea was that if you want to cancel a request, you don't use the promise api. Most of users don't cancel queries, the few people who need to cancel will have to use the callback api. That seemed to me to be the best tradeoff.

In my application I have written a wrapper around the callback api to use promises. Cancelable queries resolves with something like:
```javascript
{
  data: Promise { <pending> },
  cancel: [Function: bound cancel]
}
```
But this is more complex for users; the cancelable queries api become different from the "normal" queries; and maybe it's not the pg responsibility to do this kind of things (too high level?). And the wrapper itself needs to keep references to query...

Do you remember why returning the query was removed?